src/hotspot/share/asm/assembler.hpp

```diff
@@ -290,6 +290,18 @@ class AbstractAssembler : public ResourceObj  {
   };
 #endif
 
+  // sign-extended tolerant cast needed by callers of emit_int8 and emit_int16
+  template <typename T>
+  constexpr T narrow_cast(int x) const {
+   if (x < 0) {
+    assert(x > -(std::numeric_limits<T>::max() - 1), "too negative");
```

Comment:
```suggestion
    assert(x >= -std::numeric_limits<T>::max(), "too negative"); // >= -256 for 8 bits
```

Reply:
Thank you - I meant to check these values.